Gems/Atom/RHI/Code/Source/RHI/ShaderResourceGroupPool.cpp

```diff
@@ -246,6 +250,113 @@ namespace AZ
             return static_cast<uint32_t>(m_groupsToCompile.size());
         }
 
+        template<typename T>
+        HashValue64 ShaderResourceGroupPool::GetViewHash(AZStd::span<const RHI::ConstPtr<T>> views)
+        {
+            HashValue64 viewHash = HashValue64{ 0 };
+            for (size_t i = 0; i < views.size(); ++i)
+            {
+                if (views[i])
+                {
+                    viewHash = TypeHash64(views[i]->GetHash(), viewHash);
+                }
+            }
+            return viewHash;
+        }
+
+        template<typename T>
+        void ShaderResourceGroupPool::UpdateMaskBasedOnViewHash(
+            ShaderResourceGroup& shaderResourceGroup, Name entryName, AZStd::span<const RHI::ConstPtr<T>> views,
+            ShaderResourceGroupData::ResourceType resourceType)
+        {
+            //Get the view hash and check if it was updated in which case we need to compile those views. 
+            HashValue64 viewHash = GetViewHash<T>(views);
+            if (shaderResourceGroup.GetViewHash(entryName) != viewHash)
+            {
+                shaderResourceGroup.EnableRhiResourceTypeCompilation(static_cast<ShaderResourceGroupData::ResourceTypeMask>(AZ_BIT(static_cast<uint32_t>(resourceType))));
```

Comment:
nit: line length. Perhaps we could get a utility function to convert a resource type to a ResourceTypeMask.